### PR TITLE
Files are created in tmp dirs and they are cleaned if tests succeed (#1061)

### PR DESCRIPTION
**Pull request by GMDSantana**
As discussed in #1061, here we use this:

https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

to handle the creation of temporary directories. The files are not deleted in case there are any test failures, so they can be inspected.

### CODE REVIEW

**Comment by progval** [comment]
Thanks

Could you create a single temporary directory with the three `test-` inside it, as well as print its path? This will make it easier for people running tests to locate where the data is.

File: src/scripts/limnoria_test.py [hunk 1]
```diff
@@ -36,23 +36,28 @@
...
 
 import supybot
 import logging
 import traceback
 
 # We need to do this before we import conf.
-if not os.path.exists('test-conf'):
-    os.mkdir('test-conf')
+main_temp_dir = TemporaryDirectory()
+print(f"Temporary directory path: {main_temp_dir.name}")
```
Comment (progval) [inline src/scripts/limnoria_test.py:48]: Move this to the `.detach()` case so it's printed at the end (more visible), and only if the directory is kept.

**Comment by progval** [comment]
Thanks!
